std/format.d

```diff
@@ -3550,20 +3550,24 @@ private template hasToString(T, Char)
         enum hasToString = 0;
     }
     else static if (is(typeof(
-        {T val = void; FormatSpec!Char f; static struct S {void put(Char s){}}
-        S s; val.toString(s, f);
+        {T val = void;
+        const FormatSpec!Char f;
+        static struct S {void put(Char s){}}
+        S s;
         // force toString to take output range by ref
-        static assert(ParameterStorageClassTuple!(T.toString!(S))[0] == ParameterStorageClass.ref_);
-        static assert(ParameterStorageClassTuple!(T.toString!(S))[1] == ParameterStorageClass.ref_);
-        static assert(is(Parameters!(T.toString!(S))[1] == const));}
+        static assert( __traits(compiles, val.toString(s, f)));
```

Comment:
`static assert( __traits(compiles` is unnecessary, isn't it? Or in other words: `val.toString(s, f)` should work fine as it's already is `is(typeof(`.